source/code/plugins/oms_common.rb

```diff
@@ -11,17 +11,520 @@ class Common
+      'Africa/Tunis' => 'W. Central Africa Standard Time',
+      'Etc/GMT-1' => 'W. Central Africa Standard Time',
+      'Europe/Berlin' => 'W. Europe Standard Time',
+      'Europe/Andorra' => 'W. Europe Standard Time',
+      'Europe/Vienna' => 'W. Europe Standard Time',
+      'Europe/Zurich' => 'W. Europe Standard Time',
+      'Europe/Busingen' => 'W. Europe Standard Time',
+      'Europe/Gibraltar' => 'W. Europe Standard Time',
+      'Europe/Rome' => 'W. Europe Standard Time',
+      'Europe/Vaduz' => 'W. Europe Standard Time',
+      'Europe/Luxembourg' => 'W. Europe Standard Time',
+      'Europe/Monaco' => 'W. Europe Standard Time',
+      'Europe/Malta' => 'W. Europe Standard Time',
+      'Europe/Amsterdam' => 'W. Europe Standard Time',
+      'Europe/Oslo' => 'W. Europe Standard Time',
+      'Europe/Stockholm' => 'W. Europe Standard Time',
+      'Arctic/Longyearbyen' => 'W. Europe Standard Time',
+      'Europe/San_Marino' => 'W. Europe Standard Time',
+      'Europe/Vatican' => 'W. Europe Standard Time',
+      'Asia/Tashkent' => 'West Asia Standard Time',
+      'Antarctica/Mawson' => 'West Asia Standard Time',
+      'Asia/Oral' => 'West Asia Standard Time',
+      'Asia/Aqtau' => 'West Asia Standard Time',
+      'Asia/Aqtobe' => 'West Asia Standard Time',
+      'Indian/Maldives' => 'West Asia Standard Time',
+      'Indian/Kerguelen' => 'West Asia Standard Time',
+      'Asia/Dushanbe' => 'West Asia Standard Time',
+      'Asia/Ashgabat' => 'West Asia Standard Time',
+      'Asia/Samarkand' => 'West Asia Standard Time',
+      'Etc/GMT-5' => 'West Asia Standard Time',
+      'Pacific/Port_Moresby' => 'West Pacific Standard Time',
+      'Antarctica/DumontDUrville' => 'West Pacific Standard Time',
+      'Pacific/Truk' => 'West Pacific Standard Time',
+      'Pacific/Guam' => 'West Pacific Standard Time',
+      'Pacific/Saipan' => 'West Pacific Standard Time',
+      'Etc/GMT-10' => 'West Pacific Standard Time',
+      'Asia/Yakutsk' => 'Yakutsk Standard Time',
+      'Asia/Chita' => 'Yakutsk Standard Time',
+      'Asia/Khandyga' => 'Yakutsk Standard Time'
+    }
+
+    @@tzLocalTimePath = '/etc/localtime'
+    @@tzBaseFolder = '/usr/share/zoneinfo/'
+    @@tzRightFolder = 'right/'
 
     class << self
-      
+      def get_unified_timezoneid(filepath)
+        # remove the baseFolder path
+        tzID = filepath[@@tzBaseFolder.length..-1] if filepath.start_with?(@@tzBaseFolder)
+
+        # if the rest starts with 'right/', remove it to unify the format
+        tzID = tzID[@@tzRightFolder.length..-1] if tzID.start_with?(@@tzRightFolder)
+        
+        return tzID
+      end # end get_unified_timezoneid
+
+      def get_current_timezone
+        return @@CurrentTimeZone if !@@CurrentTimeZone.nil?
+
+        tzID = 'Unknown'
+        
+        begin
+          # looks for a file in the /usr/share/zoneinfo/, which is identical to /etc/localtime. use the file name as the timezone
```

Comment:
Move this comment lower to where it is relevant.
There could be a comment here like "if /etc/localtime is a valid symlink use the target as the current timezone."